### PR TITLE
pkg/trace: Simplify sampler updates

### PR DESCRIPTION
**Pull request by raphaelgavache**
Synchronizing adjust, decay and the generation of new sampling rates. There is no reason to decouple them and it makes modeling the behavior of the sampler hard. Test of volumes are made with them synchronized.

Updated score sampler test to match priority sampler test

### CODE REVIEW

File: pkg/trace/sampler/prioritysampler.go [hunk 1]
```diff
@@ -69,31 +69,35 @@ func NewPrioritySampler(conf *config.AgentConfig, dynConf *DynamicConfig) *Prior
 
 // Start runs and block on the Sampler main loop
 func (s *PrioritySampler) Start() {
-	s.localRates.Start()
```
Comment (Barbayar) [inline pkg/trace/sampler/prioritysampler.go:72]: Where are you calling `localRates.Start()` now? If we don't need `Sampler.Start()` anymore, let's clean it.

Comment (raphaelgavache) [reply pkg/trace/sampler/prioritysampler.go:72]: Sampler.Start is still used by score_sampler, I'll clean it up on another PR

File: pkg/trace/sampler/remote_rates.go [hunk 2]
```diff
@@ -106,57 +104,15 @@ func (r *RemoteRates) updateTPS(tpsTargets map[Signature]float64) {
...
-	defer r.mu.RUnlock()
-	for _, s := range r.samplers {
-		s.AdjustScoring()
-	}
-}
-
-// Stop stops RemoteRates main loop
-func (r *RemoteRates) Stop() {
-	close(r.exit)
-	r.stopSubscriber()
```
Comment (Barbayar) [inline pkg/trace/sampler/remote_rates.go:156]: NOTE: this can be removed after https://github.com/DataDog/datadog-agent/pull/10268

**Review by Barbayar — APPROVED** [review]

**Review by nikriek — APPROVED** [review]
love it

**Review by gbbr — APPROVED** [review]

**Comment by raphaelgavache** [comment]
I'll merge this PR after resolving conflicts from this PR https://github.com/DataDog/datadog-agent/pull/10268

**Pull request merged**